src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -242,6 +250,51 @@ private Task WriteMultiEndpointMessageAsync(ServiceMessage serviceMessage, Func<
             return Task.WhenAll(routed);
         }
 
+        private void OnAdd(HubServiceEndpoint endpoint)
+        {
+            if (!endpoint.Hub.Equals(_hubName, StringComparison.OrdinalIgnoreCase))
+            {
+                return;

```

Comment:
Can it ever happen?

Reply:
throw instead if it is defense code?

Reply:
Yes. In `ServiceEndpointManager` side endpoint is not hub distinguished, after split to hub level, it will trigger multiple times scale in Container side. So need to identify which is correct one by compare hub.